datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -317,35 +426,78 @@ func resourceDatadogSecurityMonitoringRuleCreate(ctx context.Context, d *schema.
 		return diag.FromErr(err)
 	}
 
-	d.SetId(response.SecurityMonitoringStandardRuleResponse.GetId())
+	if response.SecurityMonitoringStandardRuleResponse != nil {
+		d.SetId(response.SecurityMonitoringStandardRuleResponse.GetId())
+	} else if response.SecurityMonitoringSignalRuleResponse != nil {
+		d.SetId(response.SecurityMonitoringSignalRuleResponse.GetId())
+	} else {
+		return diag.FromErr(fmt.Errorf("SecurityMonitoringStandardRuleResponse and SecurityMonitoringSignalRuleResponse are both empty"))
+	}
+
+	return nil
+}
+
+func isSignalCorrelationSchema(d *schema.ResourceData) bool {
+	if v, ok := d.GetOk("type"); ok {
+		_, err := datadogV2.NewSecurityMonitoringSignalRuleTypeFromValue(v.(string))
+		return err == nil
+	}
+	return false
+}
 
+func checkQueryConsistency(d *schema.ResourceData) error {
+	query := d.Get("query").([]interface{})
+	signalQuery := d.Get("signal_query").([]interface{})
+	if len(query) > 0 && len(signalQuery) > 0 {
+		return fmt.Errorf("query list and signal query list cannot be both populated")
```

Comment:
I tried but it was not working since, as stated in the documentations: 
```
// Only absolute attribute paths, ones starting with top level attribute
// names, are supported. Attribute paths cannot be accurately declared
// for TypeList (if MaxItems is greater than 1), TypeMap, or TypeSet
// attributes. To reference an attribute under a single configuration block
// (TypeList with Elem of *Resource and MaxItems of 1), the syntax is
// "parent_block_name.0.child_attribute_name".
```
In our case, it is a TypeList whose MaxItems is greater than 1.